controllers/clusteroperator.go

```diff
@@ -227,3 +231,25 @@ func (r *ProvisioningReconciler) updateCOStatus(newReason StatusReason, msg, pro
 
 	return r.syncStatus(co, conds)
 }
+
+func SetCOInDisabledState(osClient osclientset.Interface, version string) error {
+	//The CVO should have created the CO
+	co, err := osClient.ConfigV1().ClusterOperators().Get(context.Background(), clusterOperatorName, metav1.GetOptions{})
+
+	if err != nil {
+		return fmt.Errorf("failed to get clusterOperator %q: %v", clusterOperatorName, err)
+	}
+
+	conds := defaultStatusConditions()
+	v1helpers.SetStatusCondition(&conds, setStatusCondition(OperatorDisabled, osconfigv1.ConditionTrue, string(ReasonUnsupported), "Nothing to do on this Platform"))
+	v1helpers.SetStatusCondition(&conds, setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, string(ReasonExpected), "Operational"))
+
+	for _, c := range conds {
+		v1helpers.SetStatusCondition(&co.Status.Conditions, c)
+	}
+	co.Status.Versions = operandVersions(version)
+	co.Status.RelatedObjects = relatedObjects()
+
+	_, err = osClient.ConfigV1().ClusterOperators().UpdateStatus(context.Background(), co, metav1.UpdateOptions{})
```

Comment:
I'm still worried about the error handling here. Do we receive an error if the ClusterOperator isn't going to be changed because it is already set with the same conditions?

Reply:
I don't see why a no-op update would error.  But maybe patch, instead of update, if you're worried about generation noise?

Reply:
SetCOInDisabledState() is called once before the start of the reconcile loop. At that time, the baremetal ClusterOperator is created by CVO using whatever is provided in /manifests. That is a bare CO with no StatusConditions, Version or RelatedObjects. So, this will be the 1st time that these values would be set.
For all other times we are updating the CO (including putting it in Disabled state) from within the reconcile loop, we use  updateCOStatus() to do that.

Reply:
I've seen errors updating other resources without changes, but maybe those were different cases. If we think this is OK, then I think this PR is probably ready to merge.

Reply:
I don't see anything unique in this function, can you just use updateCOStatus(). Note that updateCOStatus() will also need to handle the cases you talk about (no StatusConditions, Version or RelatedObjects)

Reply:
We don't have the ProvisioningReconciler object at this time so added this new method. Went down the path of making  updateCOStatus() an exported function and ended up making a lot of changes in clusteroperator.go.

Reply:
you don't have to do it this early. what i am suggesting is to do it within the SetupWithManager() function.